package.json

```diff
@@ -15,7 +15,8 @@
     "lint:eslint:fix": "eslint . --ext .js,.jsx,.ts,.tsx,.astro --fix",
     "lint:prettier": "prettier --check .",
     "lint:prettier:fix": "prettier --write .",
-    "lint:linkcheck": "RUN_LINK_CHECK=true pnpm build"
+    "lint:linkcheck": "RUN_LINK_CHECK=true pnpm build",
+    "lint:diff": ".github/scripts/lint-diff.sh"
```

Comment:
Instead of adding a script that runs all the linting checks, just define subtargets here like `lint:prettier:check-pr` that run the check commands restricted to the files changed in the current PR. The CI workflows can then use these targets instead of the more general ones it uses currently.